### PR TITLE
add comments.js

### PR DESCRIPTION
**Pull request by zrose584**
`settings` is now in the global jinja context.
You can still open the `/comments` page by using middle click.

### CODE REVIEW

File: youtube/comments.py [hunk 1]
```diff
@@ -121,10 +121,18 @@ def post_process_comments_info(comments_info):
...
+                    return "%3.1f%s" % (num, unit) if unit else num
+                num /= 1000.0
+            return "%.1f%s" % (num, 'm')
+
+        # if comment['like_count'] = '👍 ' + str(fmt_num(comment['like_count']))
         if comment['like_count'] == 1:
             comment['likes_text'] = '1 like'
         else:
-            comment['likes_text'] = str(comment['like_count']) + ' likes'
+            comment['likes_text'] = str(fmt_num(comment['like_count'])) + ' likes'
```
Comment (user234683) [inline youtube/comments.py:135]: I vote no on this. No need to trim the numbers. I think the like counts are missing commas for thousands but that can be in a separate commit/pull. Best to keep pull requests focused on just the individual change.

File: youtube/templates/comments_page.html [hunk 2]
```diff
@@ -36,7 +41,7 @@ <h2>Comments page {{ comments_info['page_number'] }}</h2>
 
         <div class="comments">
             {% for comment in comments_info['comments'] %}
-                {{ comments.render_comment(comment, comments_info['include_avatars']) }}
+                {{ comments.render_comment(comment, comments_info['include_avatars'], slim) }}
```
Comment (user234683) [inline youtube/templates/comments_page.html:44]: Remove `slim` parameter from this line since the `render_comment` function signature wasn't changed.

File: youtube/templates/watch.html [hunk 3]
```diff
@@ -14,6 +14,21 @@
...
+        border-width: 1px;
+        font-weight: bold;
+        padding-bottom: 0px;
+    }
+
+    details.replies .comment{
+        width: 600px;
+        border-style: solid;
+        border-width: 1px;
+        border-color: #0402025e;
```
Comment (user234683) [inline youtube/templates/watch.html:29]: The border does more harm than good visually. It looks better without it in my opinion.

File: youtube/templates/watch.html [hunk 4]
```diff
@@ -682,4 +697,5 @@ <h2 class="title">{{ title }}</h2>
         <script src="/youtube.com/static/js/hotkeys.js"></script>
     {% endif %}
     <script src="/youtube.com/static/js/transcript-table.js"></script>
+    <script src="/youtube.com/static/js/comments.js"></script>
```
Comment (user234683) [inline youtube/templates/watch.html:700]: If statement here would be nice based on `if settings.use_comments_js`

**Comment by user234683** [comment]
Another issue is if Youtube blocks the Tor exit node, the comment replies will be the full error page. In these cases, `fetch_url` raises `FetchError`, which has a `.code` of 429. The error bubbles up to flask. It is handled by the error handler in `__init__.py`, which returns a 502 response. So this can be fixed perhaps by catching the FetchError in comments.py and returning a minimal error if slim is True. Alternatively, the javascript could check that the xhr response is not 200 and just display a generic error.

File: youtube/comments.py (hunk 1 above)
Comment (zrose584) [reply youtube/comments.py:135]: Why? When you see a Number like 34573215 why would you care about the 215 at the end?
Also "34.6m" takes less space than "34,573,215". It's also what yt does.

File: youtube/templates/comments_page.html (hunk 2 above)
Comment (zrose584) [reply youtube/templates/comments_page.html:44]: This was intentional: Since slim is only used on /watch, timestamp_links should be added.

**Comment by user234683** [comment]
Another issue: it's not working on the /comments page (such as more comments or sort by newest) because the script isn't included there.

**Comment by zrose584** [comment]
A (minor) remaining issue: text of the replies in /comments get  passed to the "timestamps" filter.
It would be best to conditionally either set onclick="video.currentTime=X", or href="/watch?t=X".
I don't know jinja too well (I'd do it in js), so I leave this to you.

**Comment by user234683** [comment]
Sure. Once the number trimming is removed it will be ready to merge and that can be fixed
